lsp/src/saros/lsp/ui/UISynchronizerImpl.java

```diff
@@ -0,0 +1,31 @@
+package saros.lsp.ui;
+
+import java.util.concurrent.Executors;
+import saros.synchronize.UISynchronizer;
+
+/** Implementation of {@link UISynchronizer}. */
+public class UISynchronizerImpl implements UISynchronizer {
```

Comment:
If you do not correctly implement that class there is no reason to continue further development. I do not know how to access the VSC IDE UI/Main thread through JAVA. However the Jupiter Algorithm on the client side will not work correctly if the transformation is done in another thread than the thread that is responsible for changing editor contents on the client side.

If you dig down through the Github Repo you will find such a fix for Eclipse, changing the thread access (I guess it is about 10 years old).

Reply:
https://github.com/saros-project/saros/blob/master/core/src/saros/session/internal/ActivityHandler.java#L224

https://github.com/saros-project/saros/commit/ad29d64a7ece48ad172c36f4672fe5afb8c0f231